Defs/Ammo/Advanced/12x20mmCharged.xml

```diff
@@ -0,0 +1,279 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Defs>
+
+	<ThingCategoryDef>
+		<defName>Ammo12x20mmCharged</defName>
+		<label>12x20mm Charged</label>
+		<parent>AmmoAdvanced</parent>
+		<iconPath>UI/Icons/ThingCategories/CaliberCharge</iconPath>
+	</ThingCategoryDef>
+
+	<!-- ==================== AmmoSet ========================== -->
+
+	<CombatExtended.AmmoSetDef>
+		<defName>AmmoSet_12x20mmCharged</defName>
+		<label>12x20mm Charged</label>
+		<ammoTypes>
+			<Ammo_12x20mmCharged>Bullet_12x20mmCharged</Ammo_12x20mmCharged>
+			<Ammo_12x20mmCharged_AP>Bullet_12x20mmCharged_AP</Ammo_12x20mmCharged_AP>
+			<Ammo_12x20mmCharged_Ion>Bullet_12x20mmCharged_Ion</Ammo_12x20mmCharged_Ion>
+		</ammoTypes>
+		<similarTo>AmmoSet_ChargedPistol</similarTo>
+	</CombatExtended.AmmoSetDef>
+
+	<!-- ==================== Ammo ========================== -->
+
+	<ThingDef Class="CombatExtended.AmmoDef" Name="12x20mmChargedBase" ParentName="SpacerSmallAmmoBase" Abstract="True">
+		<description>Charged shot ammo used by high caliber magnum charge weapons.</description>
+		<statBases>
+			<Mass>0.21</Mass>
+			<Bulk>0.02</Bulk>
+		</statBases>
+		<tradeTags>
+			<li>CE_AutoEnableTrade</li>
+			<li>CE_AutoEnableCrafting_FabricationBench</li>
+		</tradeTags>
+		<thingCategories>
+			<li>Ammo12x20mmCharged</li>
+		</thingCategories>
+		<stackLimit>5000</stackLimit>
+	</ThingDef>
+
+	<ThingDef Class="CombatExtended.AmmoDef" ParentName="12x20mmChargedBase">
+		<defName>Ammo_12x20mmCharged</defName>
+		<label>12x20mm Charged</label>
+		<graphicData>
+			<texPath>Things/Ammo/Charged/SmallRegular</texPath>
+			<graphicClass>Graphic_StackCount</graphicClass>
+		</graphicData>
+		<statBases>
+			<MarketValue>1.21</MarketValue>
+		</statBases>
+		<ammoClass>Charged</ammoClass>
+	</ThingDef>
+
+	<ThingDef Class="CombatExtended.AmmoDef" ParentName="12x20mmChargedBase">
+		<defName>Ammo_12x20mmCharged_AP</defName>
+		<label>12x20mm Charged (Conc.)</label>
+		<graphicData>
+			<texPath>Things/Ammo/Charged/SmallConc</texPath>
+			<graphicClass>Graphic_StackCount</graphicClass>
+		</graphicData>
+		<statBases>
+			<MarketValue>1.21</MarketValue>
+		</statBases>
+		<ammoClass>ChargedAP</ammoClass>
+	</ThingDef>
+
+	<ThingDef Class="CombatExtended.AmmoDef" ParentName="12x20mmChargedBase">
+		<defName>Ammo_12x20mmCharged_Ion</defName>
+		<label>12x20mm Charged (Ion)</label>
+		<graphicData>
+			<texPath>Things/Ammo/Charged/SmallIon</texPath>
+			<graphicClass>Graphic_StackCount</graphicClass>
+		</graphicData>
+		<statBases>
+			<MarketValue>1.21</MarketValue>
+		</statBases>
+		<ammoClass>Ionized</ammoClass>
+		<generateAllowChance>0.5</generateAllowChance>
+	</ThingDef>
+
+	<!-- ================== Projectiles ================== -->
+
+	<ThingDef Name="Base12x20mmChargedBullet" ParentName="BaseBulletCE" Abstract="true">
```

Comment:
As-is, this isn't meaningfully more powerful than 10x18. Comparing across the two ammo types, regular charged: +20% bullet damage. Compared to the difference of .44 mag vs .500 S&W, where FMJ does +40% more damage.

Given that 10x18 is already punching through most flak/industrial armor and threatening power armor when using concentrated rounds, I'm not sure if there's headroom or need for another, more powerful charged magnum revolver caliber round.